concord-ui/src/components/uploadphrase.js

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import { Button } from '@rmwc/button';
+import { TextField } from '@rmwc/textfield';
+import { connect } from 'react-redux';
+import {
+  DataTable,
+  DataTableBody,
+  DataTableCell,
+  DataTableContent,
+  DataTableHead,
+  DataTableHeadCell,
+  DataTableRow
+} from '@rmwc/data-table';
+import '@rmwc/data-table/data-table.css';
+import { postUnlabelledPhrases } from '../api';
+
+export class UploadPhrase extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { reloadApiData: false, textField: '', invalidData: true, phrases: [] };
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    nextState.invalidData = !nextState.textField;
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
Will want to make this async since clearFields() is setting state also. Easiest way is -


componentWillReceieveProps(nextProps) async {
    if (nextProps.reloadApiData !== this.state.reloadApiData) {
      await this.setState({
        reloadApiData: nextProps.reloadApiData
      });
      this.clearFields();
    }
  }